internal/tasks/callback_manager_test.go

```diff
@@ -0,0 +1,512 @@
+package tasks
+
+import (
+	"encoding/json"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store/model"
+	flightlog "github.com/flightctl/flightctl/pkg/log"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+type MockPublisher struct {
+	publishedResources []ResourceReference
+}
+
+func (m *MockPublisher) Publish(payload []byte) error {
+	var resource ResourceReference
+	err := json.Unmarshal(payload, &resource)
+	if err != nil {
+		return err
+	}
+	m.publishedResources = append(m.publishedResources, resource)
+	return nil
+}
+
+func (m *MockPublisher) Close() {
+	clear(m.publishedResources)
+}
+
+var (
+	mockPublisher    *MockPublisher
+	callbacksManager CallbackManager
+)
+
+var _ = Describe("FleetUpdatedCallback", func() {
+	BeforeEach(func() {
+		mockPublisher = &MockPublisher{}
+		callbacksManager = NewCallbackManager(mockPublisher, flightlog.InitLogs())
+	})
+
+	When("both before and after are nil", func() {
+		It("does nothing", func() {
+			callbacksManager.FleetUpdatedCallback(nil, nil)
+			Expect(mockPublisher.publishedResources).To(BeEmpty())
+		})
+	})
+
+	When("before is nil and after is not nil", func() {
+		It("submits FleetValidateTask and FleetSelectorMatchTask", func() {
+			orgId := uuid.New()
```

Comment:
Nit: declare orgId above and initialize in BeforeEach

Reply:
Done.